app.js

```diff
@@ -10,8 +11,174 @@ class KNXApp extends Homey.App {
+          const availableInterfaces = this.knxInterfaceManager.getKNXInterfaceList();
+          const macs = Object.keys(availableInterfaces);
+          knxInterfaceToUse = macs.length > 0 ? availableInterfaces[macs[0]] : null;
+        }
+
+        if (!knxInterfaceToUse) {
+          this.error('Interface not found');
+        } else {
+          this.eventListenerGroupAddresses.push(`${args.interface.mac}-${args.group_address}`);
+          // Store the event listener so we can remove it later
+          knxInterfaceToUse.addKNXEventListener(args.group_address, this.onKNXEvent.bind(this, args));
+        }
+      }
+    });
+
+    this.eventListenerGroupAddresses.forEach((eventListernerId) => {
+      if (cards.filter((args) => `${args.interface}-${args.group_address}` === eventListernerId).length === 0) {
+        // TODO: figuure out how to remove the event listener, because we need the correct event listener
+        // knxInterfaceToUse.removeKNXEventListener(groupAddress, this.onKNXEvent.bind( this, args));
+      }
+    });
+  }
+
+  async interfaceAutocomplete(query, args) {
+    const results = this.knxInterfaceManager.getSimpleInterfaceList();
+    results.push({ name: '[any]', mac: 'any', ip: 'any' });
+
+    // filter based on the query
+    return results.filter((result) => (
+      result.name.toLowerCase().includes(query.toLowerCase())
+      || result.mac.toLowerCase().includes(query.toLowerCase())
+      || result.ip.toLowerCase().includes(query.toLowerCase())
+    ));
+  }
+
+  async datatypeAutocomplete(query, args) {
+    return this.availableKnxDataTypes.filter((r) => r.name.includes(query.toUpperCase()));
+  }
+
+  async sendKNXTelegram(args, state) {
+    console.log(args);
+    console.log(state);
+
+    let knxInterfaceToUse = this.knxInterfaceManager.getKNXInterface(args.interface.mac);
+    if (args.interface.mac === 'any') {
+      const availableInterfaces = this.knxInterfaceManager.getKNXInterfaceList();
+      const macs = Object.keys(availableInterfaces);
+      knxInterfaceToUse = macs.length > 0 ? availableInterfaces[macs[0]] : null;
+    }
+
+    if (!knxInterfaceToUse) {
+      return Promise.reject(new Error('Interface not found'));
+    }
+
+    if (!args.group_address) {
+      return Promise.reject(new Error('No group address selected'));
+    }
+
+    // Even if the datatype is bool the knx lib can convert from "true" to a boolean value, 
+    // so we can just pass the string, in general we let people have as much freedom as possible
+    // TODO: Add date and time support
+    let value = args.value_bool
+    if (args.value_number !== undefined)
+      value = args.value_number;
+    if (args.value_string !== undefined)
+      value = args.value_string;
+
+    if (value === undefined) {
+      return Promise.reject(new Error('No value selected'));
+    }
+
+    if (args.data_type === 'none') {
+      return knxInterfaceToUse.writeKNXGroupAddress(args.group_address, value);
```

Comment:
If you can send without datatype, Do we really need it then?

Reply:
Yes, without a datatype it will try a standard conversion from a simple type to a knx buffer. But there might be specific datatype conversions. little/big endianess and conversions between percentage and absolute values.

i.e. in KNX a value of 1 to a dimmer that represents a percentage actually means 1/255 not 1%. So the datatype has a big impact.